plugins/inventory/hypercore.py

```diff
@@ -218,19 +218,19 @@ def parse(self, inventory, loader, path, cache=False):
         super(InventoryModule, self).parse(inventory, loader, path)
         cfg = self.read_config_data(path, os.environ)
 
-        # Try getting variables from env
-        try:
-            host = os.getenv("SC_HOST")
-            username = os.getenv("SC_USERNAME")
-            password = os.getenv("SC_PASSWORD")
-            timeout = os.getenv("SC_TIMEOUT")
-        except KeyError:
+        # get variables from env
+        host = os.getenv("SC_HOST")
+        username = os.getenv("SC_USERNAME")
+        password = os.getenv("SC_PASSWORD")
+        timeout = os.getenv("SC_TIMEOUT")
+        if timeout:
+            timeout = int(timeout)
```

Comment:
So without try-catch we get quite descriptive error 
"[WARNING]:  * Failed to parse /home/polonamihalic/Workspace/Scale_computing/ans
ible_collections/scale_computing/hypercore/tests/output/.tmp/integration/invent
ory-8qguwyqy-ÅÑŚÌβŁÈ/tests/integration/targets/inventory/hypercore_inventory_an
sible_enable.yml with auto plugin: could not convert string to float: '60!'"

Anyway I Implemented try-catch as suggested, because it is better - now the error looks like this:
"[WARNING]:  * Failed to parse /home/polonamihalic/Workspace/Scale_computing/ans
ible_collections/scale_computing/hypercore/tests/output/.tmp/integration/invent
ory-kq73_qll-ÅÑŚÌβŁÈ/tests/integration/targets/inventory/hypercore_inventory_an
sible_enable.yml with auto plugin: Environ variable "SC_TIMEOUT" has invalid
value 60!. The value cannot be converted to number".

I didn't use a dedicated exception class, since I think the error description is enough (the same goes for "Missing sc_host, sc_username or sc_password)